### PR TITLE
Fix to handle getAuthToken function in sdk

### PR DESCRIPTION
**Pull request by mithunjohny**
### Description
Before the facadeChatSDK implementation introduced token validation, the omnichannel-chat-widget was able to initiate authenticated chats using the token function defined in the omnichannel-chat-sdk configuration during initialization.

However, with the new token validation in place, the getChatToken function in the SDK is not being invoked because the token fails the validity check

## Solution Proposed
Added fallback logic to check the token from authfunction in chat sdk
### Acceptance criteria
_Define what are the conditions to consider the PR has achieved the intended goal_

## Test cases and evidence
_Include what tests cases were considered, any evidence of testing for future references, to identify any corner cases, etc_

### Sanity Tests
-   [ ] You have tested all changes in Popout mode
-   [ ] You have tested all changes in cross browsers i.e Edge, Chrome, Firefox, Safari and mobile devices(iOS and Android)
-   [ ] Your changes are included in the [CHANGELOG](../CHANGE_LOG.md)

## A11y
- [ ] You have run the [Automated Accessibility Check](https://accessibilityinsights.io/docs/en/windows/getstarted/automatedchecks) and have no reported issues

__*Please provide justification if any of the validations has been skipped.*__

### CODE REVIEW

File: chat-widget/src/components/livechatwidget/common/authHelper.ts [hunk 1]
```diff
@@ -33,7 +33,16 @@ const handleAuthentication = async (chatSDK: OmnichannelChatSDK, chatConfig: Cha
             TelemetryHelper.logActionEvent(LogLevel.ERROR, { Event: TelemetryEvent.ReceivedNullOrEmptyToken });
             throw new Error(WidgetLoadCustomErrorString.AuthenticationFailedErrorString);
         }
+    } else if (chatSDK && chatSDK.chatSDKConfig?.getAuthToken) {
```
Comment (xTEddie) [inline chat-widget/src/components/livechatwidget/common/authHelper.ts:36]: It could be simplified to `chatSDK?.chatSDKConfig?.getAuthToken` instead since it's `TS`

File: chat-widget/src/components/livechatwidget/common/authHelper.ts [hunk 2]
```diff
@@ -33,7 +33,16 @@ const handleAuthentication = async (chatSDK: OmnichannelChatSDK, chatConfig: Cha
...
+        const token = await chatSDK.chatSDKConfig?.getAuthToken();
+        if (!isNullOrEmptyString(token)) {
```
Comment (xTEddie) [inline chat-widget/src/components/livechatwidget/common/authHelper.ts:38]: The `if/else` can be simplified with only one `if`

**Review by elopezanaya — APPROVED** [review]

**Review by xTEddie — APPROVED** [review]
